pkg/apis/pipeline/v1/param_types.go

```diff
@@ -208,6 +208,28 @@ func (ps Params) extractParamMapArrVals() map[string][]string {
 	return paramsMap
 }
 
+// ParseTaskandResultName parses "task name", "result name" from a Matrix Context Variable
+// Valid Example 1:
+// - Input: tasks.myTask.matrix.length
+// - Output: "myTask", ""
+// Valid Example 2:
+// - Input: tasks.myTask.matrix.ResultName.length
+// - Output: "myTask", "ResultName"
+func (p Param) ParseTaskandResultName() (string, string) {
+	if expressions, ok := p.GetVarSubstitutionExpressions(); ok {
```

Comment:
Wondering what is the case here if `ok` is false, do we fail the parsing?

Reply:
@JeromeJu We currently don't fail the parsing here we would just return empty string. I'm not sure if we would want to fail here and if there's a way to bubble up the error.